### DISCUSSION TITLE
Send signed byte array with Bleak

### DISCUSSION THREAD

**Discussion by bassepeder** [hbldh/bleak#465] [Resolved discussions] [answered]
Is it possible to send a signed byte array with Bleak?

**Comment by dlech** [comment]
Can you give more details on the use case? Maybe use something like `struct.pack()`/`struct.unpack()` to convert the data.

**Comment by bassepeder** [comment, marked as answer]
Coming from Bluetooth development in Java, I sent bytearrays like this `byte[] array = new byte[-90, 118, 92, 0, 0, -12`
I need to this in Python too, and I came across the module _array_.

You can do `array = array('b, [-90, 118, 92, 0, 0, -12])` instead of `bytearray([-90, 118, 92, 0, 0, -12])`, because bytearray in Python can only contain 0-256. 

So, I create an object with  `array = array('b, [-90, 118, 92, 0, 0, -12])` and try to do `await client_write_gatt_char(uuid_write, array)`.
This returns an MarshallingError like this: `List, Tuple, Bytearray, or Dictionary required for Dbus array.`

My question is, how can I send a **signed bytearray** with Bleak?

**Reply by dlech** [reply]
array = bytearray(array('b', [-90, 118, 92, 0, 0, -12]))

or 

    array = bytearray(array('b', [-90, 118, 92, 0, 0, -12]).tobytes())

?

**Reply by bassepeder** [reply]
Amazing, thank you! 👍🏻 💯
